### PR TITLE
Enable aggregator by default in new toy clusters

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/15381

This creates the certificates used by the aggregator, temporarily wires in the upstream namespaced roles (@openshift/security fyi), fixes discovery checks to avoid failures on missing resources, and adds a transport for the aggregator.

I can split these apart if necessary, but I wouldn't expect the review to be too bad.

@mfojtik got an "api server wiring" guy?

### CODE REVIEW

**Comment by deads2k** [comment]
@bparees fyi

**Comment by mfojtik** [comment]
@deads2k @soltysh will be my best offer, but I guess @sttts can do better with API wiring :)

**Comment by deads2k** [comment]
> @deads2k @soltysh will be my best offer, but I guess @sttts can do better with API wiring :)

Yeah, spreading the knowledge may do us some good.

**Comment by kargakis** [comment]
All of the jobs are currently broken because we have enabled approvers in the SQ => using approvers means that the SQ needs to clone the repo locally => everything breaks down because we currently have an issue with emptyDir in the cluster we run the SQ. Hold tight until it's fixed.

**Comment by kargakis** [comment]
You can still use the Rosie tags I think

**Comment by deads2k** [comment]
https://github.com/openshift/origin/issues/15467

/retest

File: pkg/cmd/server/origin/ensure.go [hunk 1]
```diff
@@ -179,6 +183,39 @@ func (c *MasterConfig) ensureComponentAuthorizationRules() {
 			glog.Errorf("Error creating bootstrap policy: %v", err)
 		}
 
+		// these are namespaced, so we can't reconcile them.  Just try to put them in until we work against rbac
```
Comment (enj) [inline pkg/cmd/server/origin/ensure.go]: Er so which role / binding are we missing exactly?

File: pkg/cmd/server/origin/aggregator.go [hunk 2]
```diff
@@ -75,6 +76,7 @@ func (c *MasterConfig) createAggregatorConfig(genericConfig genericapiserver.Con
 		ProxyClientCert:   certBytes,
 		ProxyClientKey:    keyBytes,
 		ServiceResolver:   serviceResolver,
+		ProxyTransport:    utilnet.SetTransportDefaults(&http.Transport{}),
```
Comment (deads2k) [inline pkg/cmd/server/origin/aggregator.go]: > So what exactly does this do? I traced ProxyTransport but am having a hard time grokking what is being proxied and by who and why.

It sets our "standard" defaults on a proxy transport that will be used by the aggregator to proxy requests to other API servers after we copy it and mess with dialer options.

File: pkg/cmd/server/origin/ensure.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/origin/ensure.go]: > Er so which role / binding are we missing exactly?

Several in kube-system.  One of which is related to auto-config of aggregated api servers.

**Comment by deads2k** [comment]
@enj anything else?

/retest

**Comment by enj** [comment]
/lgtm

@deads2k will resync the upstream pull once it merges.

**Comment by deads2k** [comment]
I'll make good now since the rest is lgtm'd

**Comment by deads2k** [comment]
/retest

remember to look here https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_conformance_gce/4976

**Pull request merged**